src/stories/Library/material-grid/MaterialGrid.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState } from "react";
+import {
+  RecommendedMaterial,
+  RecommendedMaterialProps,
+} from "../recommended-material/RecommendedMaterial";
+
+export type MaterialGridProps = {
+  title: string;
+  amountOfMaterialsToShow: 4 | 8 | 12 | 16 | 20 | 24 | 28 | 32;
+  materials: RecommendedMaterialProps[];
+};
+
+export const MaterialGrid: React.FC<MaterialGridProps> = ({
+  title,
+  amountOfMaterialsToShow,
+  materials,
+}) => {
+  // The JS code here is just an example of how this works.
+  // The proper implementation is in the react repo.
+  const [showAllMaterials, setShowAllMaterials] = useState(false);
+  const [displayedItemCount, setDisplayedItemCount] = useState(8);
+  const showMoreButton = materials.length > 8;
+
+  function handleSetShowAllMaterials() {
+    setShowAllMaterials(true);
+    setDisplayedItemCount(amountOfMaterialsToShow);
+  }
+
+  return (
+    <div className="material-grid">
+      <h2 className="material-grid__title">{title}</h2>
+      <ul className="material-grid__items">
+        {materials.slice(0, displayedItemCount).map((material, index) => (
+          <li key={index} className="material-grid__item">
+            <RecommendedMaterial {...material} />
+          </li>
+        ))}
+      </ul>
+      {showMoreButton && !showAllMaterials && (
+        <button
+          className="material-grid__show-more btn-primary btn-outline btn-medium"
+          data-show-more
+          aria-expanded="false"
```

Comment:
Please consider add an `aria-controls` attribute to the list.

:+1: remembering to add this `aria-expanded` attribute but [according to MDN you should also add an `aria-controls` attribute with the id of the element which is expanded when clicking the button](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded#buttons).